src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -370,13 +371,36 @@ private static void OnRequestSendingRequest(WinHttpRequestState state)
 
                 try
                 {
-                    WinHttpCertificateHelper.BuildChain(
-                        serverCertificate,
-                        remoteCertificateStore,
-                        state.RequestMessage.RequestUri.Host,
-                        state.CheckCertificateRevocationList,
-                        out chain,
-                        out sslPolicyErrors);
+                    // Create and configure the X509Chain
+                    chain = new X509Chain();
+                    chain.ChainPolicy.RevocationMode = 
+                        state.CheckCertificateRevocationList ? X509RevocationMode.Online : X509RevocationMode.NoCheck;
+                    chain.ChainPolicy.RevocationFlag = X509RevocationFlag.ExcludeRoot;
+                    // Authenticate the remote party: (e.g. when operating in client mode, authenticate the server).
+                    var serverAuthOid = new Oid("1.3.6.1.5.5.7.3.1", "1.3.6.1.5.5.7.3.1");
```

Comment:
Might be good idea to move this as readonly field of the class, since this will not change.